compiler/optimizer/LocalOpts.cpp

```diff
@@ -1166,6 +1166,7 @@ int32_t TR_BlockManipulator::performChecksAndTreesMovement(TR::Block *newBlock,
          ;
 
       TR::Block *nextBlock = prevBlock->getNextBlock();
+      // If this condition is satisfied, nextBlock != NULL
```

Comment:
I think the comment is misleading, if it's forecasting that nextBlock is non-NULL just because it is dereferenced inside the next block of code without first checking for NULL. My natural inclination is to suspect a missing NULL test rather than an undocumented non-NULL guarantee. Call me paranoid.

In this particular case, there's a condition below this code block that strongly suggests nextBlock is not necessarily non-NULL in this function. I think the condition that dereferences nextBlock to get its entry tree should actually test nextBlock != NULL first. In other words, I think you found a missing NULL test, and it should be fixed by adding the NULL test, not by adding a comment that implies the condition controlling the unguarded dereference being true means it should not be NULL.

It's still entirely possible I'm missing some subtlety here, but the NULL test doesn't hurt much if the comment you wrote is correct, whereas I think the comment is confusing if nextBlock can, in fact, be NULL (and you'd be leaving a bug in place).